README.md

```diff
@@ -79,12 +98,114 @@ For example, the incoming message contains the following body:
-For this purpose you need to specify the following SQL template:
-```$sql
-INSERT INTO itemstable(id, text, createdAt) VALUES(${id}, ${text}, current_timestamp)
+All changes will rollback, if something wrong with data.
+
+## General Sql Query Action
+
+**Expert mode**. You can execute SQL query or SQL script in this action.
+
+### Configuration field
+
+#### SQL Query 
+Put your SQL expression to `SQL Query` for further execution.
+You can put only one SQL query or several queries with delimiter `;`.
+All queries are executed in one transaction. All changes will rollback, if something wrong with one of execution.
+Also if you want to use prepared statements in your query,
+you need define prepared statement variables like this way `sqlVariableName = @MetadataVariableName:type` where:
+1. `sqlVariableName` - variable name in sql expression;
+2. `MetadataVariableName` - variable name in metadata (it can be the same as `sqlVariableName`);
+3. `type` - type of variable , following types are supported:
+   * string (also default type if type is omitted)
+   * number 
+   * boolean 
+   
+For example, for sql expression `SELECT * FROM tableName WHERE column1 = 'text' AND column2 = 15` you need to use following template:
+`SELECT * FROM tableName WHERE column1 = @column1:string AND column2 = @column2:number` and put values into generated metadata.
+
+![image](https://user-images.githubusercontent.com/16806832/53731432-635a2200-3e83-11e9-9a4e-0fc26aeeb001.png)
+
+### Input metadata
+Input metadata is generated from `SQL Query` configuration field if this field contains at list one defined value.
+
+### Output metadata
+
+Output metadata is an array of arrays with the result of query execution and depends on the count of SQL queries which were executed. 
+There is an empty array in output metadata, if execution does not return any results. 
+For example, for sql script:
+```$xslt
+INSERT INTO tableOne (column1, column2) VALUES ('value1', 'value2');
+SELECT * FROM table2;
 ```
-If something wrong with data, all changes will rollback.
+the first sql query `INSERT INTO tableOne (column1, column2) VALUES ('value1', 'value2')` does not return any values and
+the second sql query `SELECT * FROM table2` returns two records.
+Output metadata for this example is:
+
+```$xslt
+[
+  [],
+  [
+    {
+      "col2": 123,
+      "col1": "abc"
+    },
+    {
+      "col2": 456,
+      "col1": "def"
+    }
+  ]
+]
+``` 
+
+## Execute Sql Injection
+
+**Expert mode**. You can execute Sql Injection in this action.
+You can not use prepare statement there, for this purpose use **General Sql Query Action**.
+
+### Input metadata
+
+Input metadata contains one field `Sql Injection string`. You can put there SQL query, SQL script or set of SQL queries from the previous step.
+You can put only one SQL query or several queries with delimiter `;`.
+All queries are executed in one transaction. All changes will rollback, if something wrong with one of execution.
```

Comment:
Misprint:
All queries are executed in one transaction. All changes will rollback, if something wrong with one of *executions*.